docs/validators/node-upgrades.md

```diff
@@ -0,0 +1,73 @@
+---
+id: node-upgrades
+title: Elrond Node upgrades
+---
+
+## **Introduction**
+
+Once a new node's binary is ready to be deployed on one of a network (mainnet, testnet or devnet), nodes operators must 
+perform the upgrade to the newest version. These releases are always announced on Elrond [Validators chat](https://t.me/ElrondValidators) 
+plus via other communication channels, depending on the case. 
+
+## **Types of upgrades**
+
+Currently, we have the following types of upgrades:
+
+A. - **all nodes need to upgrade**:  upgrades that involve processing changes with an activation epoch (as explained below) 
+and have to be performed by all nodes operators in order to keep the same view over the network and not cause service disruptions.
+
+B. - **optional upgrades**: upgrades that for example simply add a new Rest API endpoint or improve the trie syncing timing 
+are not critical from a processing point of view, and they are optional. If the nodes operators think the new feature will help them,
+they can proceed with the upgrade without losing the compatibility with the network.
+
+C. - **only validators need to upgrade**: upgrades that, for example, include new features that only trigger validators (ratings changes, 
+transactions selection improvements and so on). Observers (nodes that don't have a stake attached) don't need to perform the upgrade.
+
+## **Activation epochs**
+
+In order to make the upgrades as smooth as possible and to ensure that each node has the same view over the network, 
+Elrond has a so called *activation epoch* mechanism that allows the node to have both versions of the code (before and 
+after the activation epoch) so that we can maintain the backwards compatibility. The system will behave accordingly to 
+the epoch.
+
+### **Deterministic time / height for upgrades**
+
+As compared to other protocols that perform upgrades that start at a specific block height, releases for Elrond nodes
+don't have a specific block height where the new updates become effective, but rather the first block in the 
+activation epoch will make the nodes proceed with the updated versions of the components.
+
+Since the height of the first block in an epoch isn't deterministic (due to possible roll-backs), the Network Height
```

Comment:
Although the phrasing `isn't deterministic` is correct, perhaps we can rewrite to `cannot be known in advance`?

Reply:
Also, above we use lower case, but here title case (`Network Height`).